File: Introduccion a POO/Problema1/Persona.java
// 1. Crea una clase Persona con nombre, edad y ciudad
public class Persona {
    private String nombre;
    private int edad;
    private String ciudad;

    public Persona(String nombre, int edad, String ciudad) {
        this.nombre = nombre;
        this.edad = edad;
        this.ciudad = ciudad;
    }

    // a) Agrega un método para mostrar el saludo: “Hola, soy {nombre} de {ciudad}”
    public void Saludo() {
        System.out.println("Hola, soy " + nombre + " de " + ciudad);
    }

    public void Verificar_edad() {
        if (edad >= 18) {
            System.out.println("Mayor de edad");
        } else {
            System.out.println("Menor de edad");
        }
    }

    public static void main(String[] args) {
        System.out.println("Programa enJava");
        System.out.println("b) Crea tres personas y muestra su saludo");
        Persona persona1 = new Persona("Crisver", 19, "El Alto");
        Persona persona2 = new Persona("Crismar", 18, "La Paz");
        Persona persona3 = new Persona("Samuel", 15, "El Alto");

        persona1.Saludo();
        persona2.Saludo();
        persona3.Saludo();

        System.out.println("c) Agrega un método para verificar si es mayor de edad");
        persona1.Verificar_edad();
        persona2.Verificar_edad();
        persona3.Verificar_edad();
    }
}

File: Introduccion a POO/Problema1/Persona.py
# 1. Crea una clase Persona con nombre, edad y ciudad
class Persona:
    def __init__(self, nombre, edad, ciudad):
        self.nombre = nombre
        self.edad = edad
        self.ciudad = ciudad
    # a) Agrega un método para mostrar el saludo: “Hola, soy {nombre} de {ciudad}”
    def Saludo(self):
        return f"Hola, soy {self.nombre} de {self.ciudad}"
    
    def Verificar_edad(self):
        if self.edad >= 18:
            return "Mayor de edad"
        else:
            return "Menor de edad"
        
print("Programa en Python")        
print("b) Crea tres personas y muestra su saludo")        
persona1 = Persona("Crisver", 19, "La Paz")
persona2 = Persona("Crismar", 18, "El Alto")
persona3 = Persona("Samuel", 15, "La Paz")
print(persona1.Saludo())
print(persona2.Saludo())
print(persona3.Saludo())
print("c) Agrega un método para verificar si es mayor de edad")
print(persona1.Verificar_edad())
print(persona2.Verificar_edad())
print(persona3.Verificar_edad())
    
